api/v1alpha1/bmctask_types.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2022 Tinkerbell.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BMCTaskConditionType represents the condition of the BMC Job.
+type BMCTaskConditionType string
+
+const (
+	// TaskCompleted represents successful completion of the BMC Task.
+	TaskCompleted BMCJobConditionType = "Completed"
+	// TaskFailed represents failure in BMC task execution.
+	TaskFailed BMCJobConditionType = "Failed"
+)
+
+// BMCTaskSpec defines the desired state of BMCTask
+type BMCTaskSpec struct {
+	// Task defines the specific action to be performed.
+	Task Task `json:"task"`
+}
+
+// Task represents the action to be performed.
+// A single task can only perform one type of action.
+// For example either PowerAction or OneTimeBootDeviceAction.
+// +kubebuilder:validation:MaxProperties:=1
+type Task struct {
+	// PowerAction represents a baseboard management power operation.
+	PowerAction *PowerAction `json:"powerAction,omitempty"`
+
+	// OneTimeBootDeviceAction represents a baseboard management one time set boot device operation.
+	OneTimeBootDeviceAction *OneTimeBootDeviceAction `json:"oneTimeBootDeviceAction,omitempty"`
+}
+
+type PowerAction struct {
+	// State represents the requested power state to set for the baseboard management.
+	// +kubebuilder:validation:Enum=PowerOn;HardPowerOff;SoftPowerOff;Status;Cycle;Reset
+	PowerControl PowerControl `json:"powerControl"`
+}
+
+type OneTimeBootDeviceAction struct {
+	// Devices represents the boot devices, in order for setting one time boot.
+	// Currently only the first device in the slice is used to set one time boot.
+	Devices []BootDevice `json:"device"`
+
+	// EFIBoot specifies to EFI boot for the baseboard management.
```

Comment:
Non-blocking: we need to clarify exactly what this does. I know it translates, under the hood, into passing `efiboot=true` in, for example, the IPMI payload but I think that behavior and result needs documenting at this level.